src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkReferenceExpression.kt

```diff
@@ -29,15 +27,22 @@ class SmkReferenceExpression(node: ASTNode): PyElementImpl(node), PsiNamedElemen
     private fun getNameNode() = getIdentifierNode(node)
 
     private class SmkRuleOrCheckpointNameReference(
-            element: PsiElement,
+            private val element: PsiNamedElement,
             textRange: TextRange
-    ) : PsiReferenceBase<PsiElement>(element, textRange) {
+    ) : PsiReferenceBase<PsiElement>(element, textRange), PsiPolyVariantReference {
         private val key: String = element.text
 
         override fun resolve(): PsiElement? =
                 getRules().firstOrNull { it.first == key }?.second ?:
                 getCheckpoints().firstOrNull { it.first == key }?.second
 
+        override fun multiResolve(incompleteCode: Boolean): Array<ResolveResult> {
+            val elements = mutableListOf<PsiElement>()
+            elements.addAll(getRules().filter { it.first == key }.map { it.second })
```

Comment:
Same as 
```
override fun multiResolve(incompleteCode: Boolean): Array<ResolveResult> =
                (getRules() + getCheckpoints())
                        .filter { (name, _) -> name == key }
                        .map { (_, psi) -> RatedResolveResult(RatedResolveResult.RATE_NORMAL, psi) }
                        .toTypedArray()
```

I prefer my version - less duplicaton and `+` already defined for lists, no need in `list.addAll()`